### DISCUSSION TITLE
Define thermal conductivity and specific heat in the HeatConductionMaterial. Thermal conductivity and specific heat are functions of temperature.

### DISCUSSION THREAD

**Discussion by Mrxiongjy** [idaholab/moose#27655] [Q&A General] [answered]
### Check these boxes if you have followed the posting rules.

- [X] Q&A General is the most appropriate section for my question
- [X] I have consulted the posting Guidelines on the Discussions front page
- [X] I have searched the Discussions forum and my question has not been asked before
- [X] I have searched the MOOSE website and the documentation does not answer my question
- [X] I have formatted my post following the posting guidelines (screenshots as a last resort, triple back quotes around pasted text)

### Question

<Your question here>
Define thermal conductivity and specific heat in the HeatConductionMaterial. Thermal conductivity and specific heat are functions of temperature.
[thermal_conductivity_1]
type = HeatConductionMaterial
temp = T
min_T = 298
# thermal_conductivity = 5
# specific_heat = 100
thermal_conductivity_temperature_function = cla_cond
specific_heat_temperature_function = cla_spe
block = 1
[]
[cla_spe]
type = PiecewiseLinear
x = '300 400 640 1090 1093 1113 1133 1153 1173 1193 1213 1233 1248 2098 2099'
y = '281 302 331 375 502 590 615 719 816 770 619 469 356 356 356'
[]
[cla_cond]
type = ParsedFunction
Expression = 7.51 + t * 2.09 * 10 ^ (2) - 1.45 * 10 ^ (5) * t ^ 2 + 7.67 * 10 ^ (9) * t ^ 3
[]
[]
In the [cla_cond] equation, it seems that t here is no longer the time variable, does t variable represent the temperature variable? Is this way feasible?

**Comment by Mrxiongjy** [comment]
Define thermal conductivity and specific heat in the HeatConductionMaterial. Thermal conductivity and specific heat are functions of temperature.
```
[thermal_conductivity_1]
type = HeatConductionMaterial
temp = T
min_T = 298
# thermal_conductivity = 5
# specific_heat = 100
thermal_conductivity_temperature_function = cla_cond
specific_heat_temperature_function = cla_spe
block = 1
[]
[cla_spe]
type = PiecewiseLinear
x = '300 400 640 1090 1093 1113 1133 1153 1173 1193 1213 1233 1248 2098 2099'
y = '281 302 331 375 502 590 615 719 816 770 619 469 356 356 356'
[]
[cla_cond]
type = ParsedFunction
Expression = 7.51 + t * 2.09 * 10 ^ (2) - 1.45 * 10 ^ (5) * t ^ 2 + 7.67 * 10 ^ (9) * t ^ 3
[]
[]
```
In the [cla_cond] equation, it seems that t here is no longer the time variable, does t variable represent the temperature variable? Is this way feasible?

**Reply by GiudGiud** [reply, marked as answer]
t now representes the tiemperature variable when it is used in a HeatConductionMaterial.

To still have time in the equation, you need to do a trick, add this:
```
[Postprocessors]
  [time_pp]
    type = ParsedPostprocessor
    expression = 't'
    use_t = true
  []
[]
```
then in the ParsedFunction for cla_cond, whenever you need `time` instead of temperature, instead of using `t`, use `time_pp`
Remember that you have to declare it in the symbols_names (you can use anything you want there to represent the `time_pp` pp) and values (you have to refer to the `time_pp` there)

**Reply by Mrxiongjy** [reply]
Thank you very much

**Reply by Mrxiongjy** [reply]
@GiudGiud Dear expert 
I want to develop a simple nuclear fuel performance analysis program similar to Bison, do you know about using moose to develop such a program?

**Reply by GiudGiud** [reply]
Hello
It's certainly possible since that's how Bison is made. You can use the Bison documentation to help 
https://mooseframework.inl.gov/bison/

**Reply by Mrxiongjy** [reply]
@GiudGiud  ∆ρ , the total amount of densification that can occur as a percent of fuel theoretical density，I'm not sure what the value of this maximum density change is, and the Bison Theory manual doesn't give it, do you know？
![1](https://github.com/idaholab/moose/assets/113361478/d6ba4158-356d-43b4-a2f3-23d7d6e05954)

**Reply by GiudGiud** [reply]
Hello

You'll have to find this in the open litterature. Use something reasonable you may find for another fuel type until you find the real value

Guillaume

**Reply by Mrxiongjy** [reply]
thanks
